src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -26,15 +30,15 @@
      */
     public static final String DEFAULT_READ_NAME_REGEX = "<optimized capture of last three ':' separated fields as numeric values>".intern();
 
-    private final int[] tmpLocationFields = new int[3]; // for optimization of addLocationInformation
+    private transient final int[] tmpLocationFields = new int[3]; // for optimization of addLocationInformation
```

Comment:
I know it's probably unecessary to serialize this because it's a tmp field, but this is not a commonly serialized thing so shaving a few bytes off isn't worth the complexity of unnecessary transitivity.   I would make this non-transient, or if it must be transient I believe it has to be reinitialized in the deserializer.